### ISSUE TITLE
maybe(T) gets flow type T instead of ?T

### ISSUE THREAD

**Issue by dbrockman** [andreypopp/validated#6]
First of all, this is a really cool use of Flow!

The only problem I've had so far is that `maybe()` will have the flow type of the type it wraps, i.e. `maybe(string)` will have the flow type `string` and not `?string`

I believe that this can be fixed by changing the definition of `MaybeNode<V>` so that it extends `Node<?V>` instead of `Node<$NonMaybeType<V>>`

```diff
+ export class MaybeNode<V> extends Node<?V> {
- export class MaybeNode<V> extends Node<$NonMaybeType<V>> {
```

**Comment by andreypopp** [comment]
Good catch! Will fix asap. Or if you want to do a PR ;) (tests could be also useful)

**Comment by andreypopp** [comment]
Fixed in 0.8.3
